### PR TITLE
Allow users to spawn docker and run tests

### PR DESCRIPTION
**Pull request by ssh24**
Dockerode script is good for CI, but it cleans up after the test
The script keeps the docker container alive and allows to debug easily on test failures

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+     PASSWORD=$4		
+ fi		
+ if [ "$5" ]; then		
+     DATABASE=$5		
+ fi		
+ 		
+ ## check if docker exists		
+ printf "\n${RED}>> Checking for docker${PLAIN} ${GREEN}...${PLAIN}"		
+ docker -v > /dev/null 2>&1		
+ DOCKER_EXISTS=$?
```
Comment (b-admike) [inline setup.sh]: Check for `docker-compose` as well since we use it in this script or remove the dependency on it and this can be the same.

Comment (ssh24) [inline setup.sh]: Good catch. Fixed.

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+     PASSWORD=$4		
+ fi		
+ if [ "$5" ]; then		
+     DATABASE=$5		
+ fi		
+ 		
+ ## check if docker exists		
+ printf "\n${RED}>> Checking for docker${PLAIN} ${GREEN}...${PLAIN}"		
+ docker -v > /dev/null 2>&1
+ docker-compose -v > /dev/null 2>&1
```
Comment (b-admike) [inline setup.sh]: If `$DOCKER_EXISTS` checks for both the output of docker and docker-compose checks then this is good. If not, I suggest you have another var `$DOCKER_COMPOSE_EXISTS` for it separately and check for both below

Comment (ssh24) [inline setup.sh]: Fixed.

**Pull request merged**